Ability to compile the clarity vm to wasm

lgalabru opened the pull request.

Commit:
chore: update rusqlite

Commit:
fix: remove sha2 = ["asm"] from lib

jcnelson commented on stacks-common/Cargo.toml:
```diff
@@ -66,9 +67,3 @@ default = ["developer-mode"]
 testing = []
-
-[target.'cfg(all(any(target_arch = "x86_64", target_arch = "x86", target_arch = "aarch64"), not(target_env = "msvc")))'.dependencies]
-sha2 = { version = "0.10", features = ["asm"] }
```
We need this for performance.  The MARF is noticeably slower without it, last I checked.

zone117x replied:
Might be worth checking if the `sha2` crate is smart enough now to ignore the `asm` feature on unsupported targets -- if so then `features = ["asm"]` can be used without messy the target cfg. It was still required earlier this year though.

lgalabru replied:
Per the `sha2` documentation, developers should avoid enabling the `asm` feature in libraries (which stacks-common is). I re-introduced the `asm` feature at the  `stacks-node` level in https://github.com/stacks-network/stacks-blockchain/pull/3293/commits/7922ec0d91e0e1a623b6e11e9e0cd2c821bb3014

Commit:
chore: strip unused import

Commit:
chore: feature flag vrf

VRF could be compiled to WASM if we were using v1.0.1 instead of 1.0.0.pre (tested)

Commit:
chore: feature flag log

logs + associate crates are not desirable in the context of a library

Commit:
chore: condition use of libsecp256k1

when compiling to wasm, pull a pure rust implementation of secp256k1

Commit:
chore: feature flag sqlite

Commit:
chore: tweak default feature flags

Commit:
fix: leftovers

Commit:
Merge branch 'next' into chore/update-rusqlite

# Conflicts:
#	src/clarity_vm/database/mod.rs

Commit:
chore: reintroduce sha2-asm at the stacks-node level

Commit:
chore: cargo fmt

Commit:
fix: logging macros

Commit:
fix: make make_keyword_reference public

Commit:
fix: re-export rusqlite

Commit:
fix: enable developer-mode for wasm

Commit:
chore: tweak logs approach

jcnelson committed:
Merge branch 'next' into chore/update-rusqlite

Commit:
feat: improve ExecutionResult model

Commit:
Merge branch 'chore/update-rusqlite' of github.com:blockstack/stacks-blockchain into chore/update-rusqlite

Commit:
Merge branch 'next' into chore/update-rusqlite

Commit:
Merge branch 'next' into chore/update-rusqlite

# Conflicts:
#	clarity/src/vm/analysis/mod.rs
#	clarity/src/vm/ast/parser/v1.rs
#	clarity/src/vm/docs/contracts.rs
#	clarity/src/vm/tests/assets.rs
#	clarity/src/vm/tests/contracts.rs
#	clarity/src/vm/tests/traits.rs
#	src/chainstate/stacks/boot/contract_tests.rs
#	src/clarity_vm/clarity.rs
#	src/core/mempool.rs

jcnelson commented:
N.B. we'll wait on reviewing and merging this until after 2.1.

Commit:
fix: rusqlite migration

Commit:
chore: cargo fmt

obycode committed:
feat: set an appropriate span for desugared tuples

Commit:
Merge pull request #3421 from stacks-network/feat/sugar-tuple-spans

feat: set an appropriate span for desugared tuples

Commit:
Merge branch 'next' into chore/update-rusqlite

# Conflicts:
#	clarity/src/vm/analysis/trait_checker/tests.rs
#	clarity/src/vm/analysis/type_checker/mod.rs
#	clarity/src/vm/analysis/type_checker/natives/options.rs
#	clarity/src/vm/analysis/type_checker/tests/contracts.rs
#	clarity/src/vm/analysis/type_checker/tests/mod.rs
#	clarity/src/vm/callables.rs
#	clarity/src/vm/contexts.rs
#	clarity/src/vm/functions/mod.rs
#	clarity/src/vm/types/serialization.rs
#	src/burnchains/db.rs
#	src/clarity_vm/database/mod.rs
#	testnet/stacks-node/src/tests/epoch_21.rs

Commit:
Merge branch 'chore/update-rusqlite' of github.com:blockstack/stacks-blockchain into chore/update-rusqlite

Commit:
fix: broken test

Commit:
patch: vm issue #1

Commit:
patch: vm issue #2

jcnelson committed:
Merge branch 'next' into chore/update-rusqlite

lgalabru commented:
Closing this PR - to be reworked in https://github.com/stacks-network/stacks-blockchain/pull/3478.
